internal/prepare/prepare_test.go

```diff
@@ -0,0 +1,67 @@
+package prepare_test
+
+import (
+	"testing"
+
+	"github.com/canonical/sqlair/internal/parse"
+	"github.com/canonical/sqlair/internal/prepare"
+	. "gopkg.in/check.v1"
+)
+
+// Hook up gocheck into the "go test" runner.
+func Test(t *testing.T) { TestingT(t) }
+
+type ParserSuite struct{}
+
+var _ = Suite(&ParserSuite{})
+
+type Address struct {
+	ID       int    `db:"id"`
+	District string `db:"district"`
+	Street   string `db:"street"`
+}
+
+type Person struct {
+	ID         int    `db:"id"`
+	Fullname   string `db:"name"`
+	PostalCode int    `db:"address_id"`
+}
+
+func (s *ParserSuite) TestValidPrepare(c *C) {
+	testList := []struct {
+		input            string
+		prepareArgs      []any
+		expectedPrepared string
+	}{{
+		"SELECT street FROM t WHERE x = $Address.street",
+		[]any{Address{}},
+		"SELECT street FROM t WHERE x = ?",
+	}, {
+		"SELECT p FROM t WHERE x = $Person.id",
+		[]any{Person{}},
+		"SELECT p FROM t WHERE x = ?",
+	}}
+	for _, test := range testList {
+		parser := parse.NewParser()
+		parsedExpr, _ := parser.Parse(test.input)
+		preparedExpr, err := prepare.Prepare(parsedExpr, test.prepareArgs...)
+		c.Log(err)
+		c.Assert(preparedExpr.SQL, Equals, test.expectedPrepared)
+	}
+}
+
+func (s *ParserSuite) TestMismatchedInputStructName(c *C) {
+	sql := "SELECT street FROM t WHERE x = $Address.street"
+	parser := parse.NewParser()
+	parsedExpr, err := parser.Parse(sql)
+	_, err = prepare.Prepare(parsedExpr, Person{ID: 1})
+	c.Assert(err, ErrorMatches, "cannot prepare expression: unknown type: Address")
+}
+
+func (s *ParserSuite) TestMissingTagInput(c *C) {
+	sql := "SELECT street FROM t WHERE x = $Address.number"
+	parser := parse.NewParser()
```

Comment:
It does feel off, I'm currently reading my way through the google style guide and it suggests using something like `parse.New` to avoid repetition. At some point we're going to do a big PR that makes it conform to the  The `typeinfo` module was originally `reflect` which made more sense, but clashed with the standard library.

Reply:
Ah right, I understand.

I actually really like the idea of using `parse.New` (or something similar), it feels very nice and concise.